lib/spack/spack/build_systems/cuda.py

```diff
@@ -59,8 +61,8 @@ def cuda_flags(arch_list):
 
     depends_on('cuda@5.0:10.2', when='cuda_arch=30')
     depends_on('cuda@5.0:10.2', when='cuda_arch=32')
-    depends_on('cuda@5.0:10.2', when='cuda_arch=35')
-    depends_on('cuda@6.5:10.2', when='cuda_arch=37')
+    depends_on('cuda@5.0:',     when='cuda_arch=35')
+    depends_on('cuda@6.5:',     when='cuda_arch=37')
 
     depends_on('cuda@6.0:',     when='cuda_arch=50')
```

Comment:
SMs 3.5-5.0 are deprecated, but not yet dropped in CUDA 11 (wikipedia page on this is too conservative)